src/main/java/org/spongepowered/common/data/util/NbtDataUtil.java

```diff
@@ -196,6 +198,31 @@ public static void setLoreToNBT(ItemStack stack, List<Text> lore) {
         stack.getSubCompound(ITEM_DISPLAY, true).setTag(ITEM_LORE, list);
     }
 
+    public static boolean hasColorFromNBT(ItemStack stack) {
+        return !stack.hasTagCompound() ||
+                !stack.getTagCompound().hasKey(ITEM_DISPLAY) ||
+                !stack.getTagCompound().getCompoundTag(ITEM_DISPLAY).hasKey(ITEM_COLOR);
+    }
```

Comment:
remove the ! and replace the || with && 
else it would have a color if the stack doesn't have a tagCompound